pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -372,6 +372,19 @@ def test_datetime_invalid_datatype(self):
         with pytest.raises(TypeError):
             pd.to_datetime(pd.to_datetime)
 
+    @pytest.mark.parametrize('date, format',
+                             [('2017-20', '%Y-%W'),
+                              ('20 Sunday', '%W %A'),
+                              ('20 Sun', '%W %a'),
+                              ('2017-21', '%Y-%U'),
+                              ('20 Sunday', '%U %A'),
+                              ('20 Sun', '%U %a')])
+    def test_week_without_day_and_calendar_year(self, date, format):
+        # GH16774
+
+        with pytest.raises(ValueError):
```

Comment:
Instead of `pytest.raises`, use `tm.assert_raises_regex` to check for the error message as well.

Reply:
Thanks. Done.